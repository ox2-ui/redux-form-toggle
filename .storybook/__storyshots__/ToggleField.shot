exports[`default`] = `
<label
  className="dEWRtq"
  style={undefined}>
  <div
    className="cFNsmu"
    disabled={undefined}
    style={undefined}>
    my label
  </div>
  <div
    className="ddmShN">
    <input
      checked={true}
      className="lmkVLy"
      color="#2ecc71"
      disabled={undefined}
      onChange={[Function]}
      type="checkbox" />
    <span
      className="iZgbnP">
      <i
        className="bqdnEt" />
    </span>
  </div>
</label>
`;
