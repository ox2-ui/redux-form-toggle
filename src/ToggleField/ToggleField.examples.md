### ToggleField

    // Redux stub START
    const input = {
      value: true,
      onChange: (value) => console.log('🍇', value), // eslint-disable-line no-console
    };
    // Redux stub END

    <ToggleField label="My label" input={input} />
