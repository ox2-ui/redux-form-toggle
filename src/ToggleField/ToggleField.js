import React, { PropTypes } from 'react';
import Toggle from '@ox2/toggle/Toggle';

/**
 * ToggleField Component
 */
const ToggleField = ({ input, ...custom }) => (
  <Toggle
    value={input.value ? true : false} // eslint-disable-line no-unneeded-ternary
    onChange={input.onChange} // eslint-disable-line react/jsx-handler-names
    {...custom}
  />
);

ToggleField.propTypes = {
  /**
   * [redux-form] The props under the input key are what connects your input component to Redux and are meant to be destructured into your `<input/>` component.
   */
  input: PropTypes.object,
};

export default ToggleField;
