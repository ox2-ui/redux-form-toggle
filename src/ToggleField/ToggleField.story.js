import React from 'react';
import { storiesOf } from '@kadira/storybook';
import ToggleField from './ToggleField';

const input = {
  value: true,
  onChange: (value) => console.log('🍇', value), // eslint-disable-line no-console
};

storiesOf('ToggleField', module)
  .add('default', () => (
    <ToggleField
      input={input}
      label="my label"
    />
  ));
