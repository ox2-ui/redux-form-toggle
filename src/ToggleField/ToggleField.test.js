/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ToggleField from './ToggleField';

const input = {
  value: true,
  onChange: (value) => console.log('🍇', value), // eslint-disable-line no-console
};

it('renders correctly', () => {
  const wrapper = mount(
    <ToggleField
      input={input}
      label="my label"
    />
  );

  expect(wrapper).toMatchSnapshot();
});
