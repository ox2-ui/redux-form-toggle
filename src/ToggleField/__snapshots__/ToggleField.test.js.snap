exports[`test renders correctly 1`] = `
<ToggleField
  input={
    Object {
      "onChange": [Function],
      "value": true,
    }
  }
  label="my label">
  <Toggle
    label="my label"
    onChange={[Function]}
    value={true}>
    <styled.label>
      <label
        className="fhQNhR">
        <styled.div>
          <div
            className="hGyXZB">
            my label
          </div>
        </styled.div>
        <styled.div>
          <div
            className="ddmShN">
            <styled.input
              checked={true}
              color="#2ecc71"
              onChange={[Function]}
              type="checkbox">
              <input
                checked={true}
                className="lmkVLy"
                color="#2ecc71"
                onChange={[Function]}
                type="checkbox" />
            </styled.input>
            <styled.span>
              <span
                className="iZgbnP">
                <styled.i>
                  <i
                    className="bqdnEt" />
                </styled.i>
              </span>
            </styled.span>
          </div>
        </styled.div>
      </label>
    </styled.label>
  </Toggle>
</ToggleField>
`;
